src/index.d.ts

```diff
@@ -1,14 +1,23 @@
 // Import Third-party Dependencies
-import { Maintainer } from "@npm/types";
 import { Scanner } from "@nodesecure/scanner";
 
-export function extractAllAuthorsFromLibrary(library: Scanner.Payload, flags: flagAuthor[] | null): authorsResponse
+export function extractAllAuthorsFromLibrary(library: Scanner.Payload, opts: options): Promise<authorsResponse[]>
+
+export interface options {
+  flags: flagAuthor[],
+  domainInformations: boolean,
+}
 
 export interface authorsResponse {
-  authors: Maintainer[] | [],
-  flags: flagAuthor[] | []
+  name?: string;
+  email?: string;
+  url?: string;
+  flagged: boolean,
+  packages: unknown[],
+  expirationDate: string,
+  mxRecords: unknown[],
```

Comment:
Shouldn't be optional keys? And to me, it would be better if those keys were in a nested object as `domain`?